tests/main/xdg-runtime-dir/task.yaml

```diff
@@ -0,0 +1,17 @@
+summary: Ensure that XDG_RUNTIME_DIR is created
+
+details: |
+    Ensure that the XDG_RUNTIME_DIR is created when a snap is launched.
+
+restore: |
+    snap remove --purge test-snapd-xdg-dir || true
+
+execute: |
+    echo Testing XDG_RUNTIME_DIR
+    snap install --edge test-snapd-xdg-dir
+    rm -f /run/user/0/snap.test-snapd-xdg-dir
```

Comment:
Perhaps move this call to the `prepare:` section, so it's clear it is setting the environment we expect the test to be run in. We don't expect the `snap install` call to create this directory, right?

Reply:
Done.